File: app/src/main/java/com/sakuna63/tumblrin/application/di/module/ActivityModule.java
package com.sakuna63.tumblrin.application.di.module;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;

import com.sakuna63.tumblrin.application.di.scope.ActivityScope;
import com.trello.rxlifecycle.LifecycleTransformer;
import com.trello.rxlifecycle.components.support.RxAppCompatActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {
    private final RxAppCompatActivity activity;

    public ActivityModule(RxAppCompatActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    Activity activity() {
        return activity;
    }

    @Provides
    @ActivityScope
    Context context() {
        return activity;
    }

    @Provides
    @ActivityScope
    LayoutInflater layoutInflater() {
        return activity.getLayoutInflater();
    }

    @Provides
    @ActivityScope
    LifecycleTransformer lifecycleTransformer() {
        return activity.bindToLifecycle();
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/di/module/ActivityModule.kt
package com.sakuna63.tumbin.application.di.module

import com.sakuna63.tumbin.application.di.scope.ActivityScope
import com.trello.rxlifecycle.LifecycleProvider
import com.trello.rxlifecycle.android.ActivityEvent
import com.trello.rxlifecycle.components.support.RxAppCompatActivity
import dagger.Module
import dagger.Provides

@Module
class ActivityModule(private val activity: RxAppCompatActivity) {
    @Provides
    @ActivityScope
    fun LifecycleProvider(): LifecycleProvider<ActivityEvent> = activity
}
